include/nbl/builtin/glsl/shapes/rectangle.glsl

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef _NBL_BUILTIN_GLSL_SHAPES_RECTANGLE_INCLUDED_
+#define _NBL_BUILTIN_GLSL_SHAPES_RECTANGLE_INCLUDED_
+
+#include <nbl/builtin/glsl/math/functions.glsl>
+
+struct SphRect {
+    mat3 basis;
+    vec3 r0;
+    vec3 r1;
+};
+
+SphRect nbl_glsl_shapes_getSphericalRectangle(in vec3 start, in vec3 ex, in vec3 ey, in vec3 origin) 
```

Comment:
rename `start` to `observer` and `origin` to `rectangleOrigin`

Reply:
👌

Reply:
as per our discord conversation, the function signature should be `(in vec3 observer, in vec3 rectangleOrigin, in mat3 rectangleNormalBasis, in vec2 rectangleExtents)`

You can make an overload which doesn't take the `rectangleExtents` and extracts them from `rectangleBasis` while normalizing `rectangleBasis`